### ISSUE TITLE
Phoenix.reload causes a crash

### ISSUE THREAD

**Issue by metakirby5** [kasper/phoenix#169]
- Version: 2.5.1 (60)
- macOS: 10.12.3 (16D32)

This only happens sometimes - I'll post again when I can consistently reproduce.

Keybinding:

```coffee
Key.on 'r', ['cmd', 'alt', 'ctrl'], -> Phoenix.reload()
```

Console log:

```
default	19:04:00.386696 -0800	Phoenix	subsystem: com.apple.AppKit, category: General, enable_level: 0, persist_level: 0, default_ttl: 0, info_ttl: 0, debug_ttl: 0, generate_symptoms: 0, enable_oversize: 1, privacy_setting: 0, enable_private_data: 0
error	19:04:00.386739 -0800	Phoenix	An uncaught exception was raised

error	19:04:00.386793 -0800	Phoenix	<private>
error	19:04:00.386847 -0800	Phoenix	<private>
default	19:04:00.386925 -0800	Phoenix	*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[NSConcreteTask terminationStatus]: task still running'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff83d1ee7b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff98908cad objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff83d9d99d +[NSException raise:format:] + 205
	3   Foundation                          0x00007fff8574f7ec -[NSConcreteTask terminationStatus] + 189
	4   Phoenix                             0x000000010fd62499 +[NSTask(PHExtension) outputFromLaunchedTaskWithEnvironment:arguments:error:] + 713
	5   Phoenix                             0x000000010fd620e1 +[NSTask(PHExtension) searchPath] + 225
	6   Phoenix                             0x000000010fd558ef +[PHShebangPreprocessor process:atPath:error:] + 287
	7   Phoenix                             0x000000010fd4ce00 -[PHContext loadScript:] + 240
	8   Phoenix
default	19:04:00.399257 -0800	kernel	AMFI: allowing exception handler for 'Phoenix' (46476) because the process is not restricted.
```

**Comment by kasper** [comment]
Could this now be caused by your fix in #168? Seems related.

**Comment by metakirby5** [comment]
It's very probable, given that the stack trace goes through `PHShebangPreprocessor`.

I think it might be a race condition from `task.terminationStatus` being accessed prior to `[task waitUntilExit]`. I might know how to fix it, though, so I'll report back in a bit.

**Comment by metakirby5** [comment]
Seems solid so far; changes are in latest commit of #168.

**Comment by kasper** [comment]
Fixed in #198.
